### PR TITLE
[15.0][IMP] base_user_rule: added an action to group groups into a role

### PR DESCRIPTION
**Pull request by ArnauCForgeFlow**
image

### CODE REVIEW

**Review by legalsylvain — APPROVED** [review]
Nice improvement ! Thanks ! 
Could you add a test. 
Otherwise lgtm.

**Comment by ArnauCForgeFlow** [comment]
@legalsylvain Done!

File: base_user_role/wizards/group_groups_into_role.py [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+from odoo import fields, models
+
+
+class GroupGroupsIntoRole(models.TransientModel):
+    """
+    This wizard is used to group different groups into a role.
+    """
+
+    _name = "group.groups.into.role"
```
Comment (legalsylvain) [inline base_user_role/wizards/group_groups_into_role.py]:
```suggestion
    _name = "wizard.groups.into.role"
```

I think it's great if the name of the model is explicit and mention that it is transient. I think it's a norm in the odoo / oca modules, don't you think ?

File: base_user_role/wizards/group_groups_into_role.py [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    _description = "Group groups into a role"
+    name = fields.Char(
+        required=True,
+        help="Group groups into a role and specify a name for this role",
+    )
+
+    def create_role(self):
+        selected_group_ids = self.env.context.get("selected_group_ids", [])
```
Comment (legalsylvain) [inline base_user_role/wizards/group_groups_into_role.py]:
```suggestion
        selected_group_ids = self.env.context.get("active_ids", [])
```

should work, don't you think ? (and avoid to add extra context in the action).

**Review by legalsylvain — COMMENTED** [review]
a minor remark, regarding naming.

**Comment by ArnauCForgeFlow** [comment]
Check now @legalsylvain

File: base_user_role/wizards/group_groups_into_role.xml [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+            action = {
+                'name': 'Specify a name for your new role',
+                'type': 'ir.actions.act_window',
+                'res_model': 'wizard.groups.into.role',
+                'target': 'new',
+                'views':[(env.ref('base_user_role.group_groups_into_role_wiz_view').id, 'form')],
+                'context': env.context
+            }
+        </field>
+    </record>
```
Comment (legalsylvain) [inline base_user_role/wizards/group_groups_into_role.xml]: I don't see the interest to use an ir.actions.server. Usually odoo implements ir.actions.act_window. something like : 


    <record id="action_wizard_groups_into_role" model="ir.actions.act_window">
        <field name="name">Create Role<field>
        <field name="type">ir.actions.act_window</field>
        <field name="res_model">wizard.groups.into.role</field>
        <field name="view_type">form</field>
        <field name="view_mode">form</field>
        <field name="target">new</field>
        <field name="binding_model_id" ref="base.model_res_groups" />
    </record>

That way, no need to specify views. And it avoid to have untested python code in server action.  IMO Server action should be used only when we want to execute it via UI that is not the case here.

**Review by legalsylvain — APPROVED** [review]
nitpicking : 
```
git mv base_user_role/wizards/group_groups_into_role.xml base_user_role/wizards/wizard_groups_into_role.xml
git mv base_user_role/wizards/group_groups_into_role.py base_user_role/wizards/wizard_groups_into_role.py
```

otherwise, LGTM. Thanks the changes and your reactivity.

**Review by legalsylvain — APPROVED** [review]

**Review by JordiBForgeFlow — APPROVED** [review]
:+1: Functional and technical review.

**Comment by legalsylvain** [comment]
Thanks for this work ! Nice addition.

/ocabot merge minor

@ArnauCForgeFlow : Could you port this patch in V16 ? 
@rven : Could you cherry-pick this patch in #253 ?

Thanks.

**Comment by rven** [comment]
> Thanks for this work ! Nice addition.
> 
> /ocabot merge minor
> 
> @ArnauCForgeFlow : Could you port this patch in V16 ? @rven : Could you cherry-pick this patch in #253 ?
> 
> Thanks.

@legalsylvain Done, I squashed the different commits into 1 meaningful one!

**Pull request merged**